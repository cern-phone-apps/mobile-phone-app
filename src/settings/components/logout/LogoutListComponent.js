import React from 'react';
import PropTypes from 'prop-types';
import { ListItem } from 'react-native-elements';
import { Alert } from 'react-native';

const disconnectUserAction = async (logout, navigation, phoneService) => {
  console.log(`Disconnecting user`);
  phoneService.disconnectUser();
  logout();
  navigation.navigate('Auth');
};

const logoutFunc = (logout, navigation, phoneService) => {
  Alert.alert('Log out?', 'Are you sure you want to log out?', [
    {
      text: 'Cancel'
    },
    {
      text: 'Log Out',
      onPress: () => {
        disconnectUserAction(logout, navigation, phoneService);
      }
    }
  ]);
};

const LogoutListComponent = ({ logout, navigation, phoneService }) => {
  return (
    <ListItem
      testID="LogoutListComponent"
      key="logout"
      title="Log out"
      leftIcon={{ name: 'logout', type: 'material-community' }}
      onPress={() => logoutFunc(logout, navigation, phoneService)}
      bottomDivider
    />
  );
};

LogoutListComponent.propTypes = {
  logout: PropTypes.func.isRequired,
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  phoneService: PropTypes.shape({
    disconnectUser: PropTypes.func.isRequired
  }).isRequired
};

export default LogoutListComponent;
