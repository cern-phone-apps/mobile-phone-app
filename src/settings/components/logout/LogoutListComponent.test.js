import React from 'react';
import { render } from '@testing-library/react-native';
import LogoutListComponent from './LogoutListComponent';

const phoneService = {
  disconnectUser: jest.fn(() => {})
};

const navigation = {
  navigate: jest.fn(text => {
    expect(text).toEqual('Auth');
  })
};

const logout = jest.fn(() => {});
/*
  We render the component
*/
test('Render the component', async () => {
  const { baseElement, queryByTestId } = render(
    <LogoutListComponent
      logout={logout}
      navigation={navigation}
      phoneService={phoneService}
    />
  );

  expect(queryByTestId('LogoutListComponent')).not.toBeNull();
  expect(baseElement).toMatchSnapshot();
});
