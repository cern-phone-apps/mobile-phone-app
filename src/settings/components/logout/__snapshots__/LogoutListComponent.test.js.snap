// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      testID="LogoutListComponent"
      underlayColor="black"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 0.5,
            "borderColor": "#bcbbc1",
            "flexDirection": "row",
            "padding": 14,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": null,
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ÔçÉ
            </Text>
          </View>
        </View>
        <View
          width={16}
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "fontSize": 17,
              }
            }
            testID="listItemTitle"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            Log out
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  </View>
</View>
`;
